packages/core/src/command/hl7v2-subscriptions/hl7v2-roster-generator.ts

```diff
@@ -0,0 +1,385 @@
+import { MetriportError, errorToString, executeWithNetworkRetries } from "@metriport/shared";
+import axios from "axios";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { Client, SFTPWrapper } from "ssh2";
+import { Hl7v2Subscriber, Hl7v2Subscription } from "../../domain/patient-settings";
+import { S3Utils, storeInS3WithRetries } from "../../external/aws/s3";
+import { capture, out } from "../../util";
+import { Config } from "../../util/config";
+import { CSV_FILE_EXTENSION, CSV_MIME_TYPE } from "../../util/mime";
+import {
+  Hl7v2RosterConfig,
+  Hl7v2SubscriberApiResponse,
+  Hl7v2SubscriberParams,
+  SftpConfig,
+} from "./get-subscribers";
+
+dayjs.extend(duration);
+
+const region = Config.getAWSRegion();
+
+type SubscriberRecord = Record<string, string>;
+
+type RosterGenerateProps = {
+  config: Hl7v2RosterConfig;
+  bucketName: string;
+  apiUrl: string;
+};
+
+const HL7V2_SUBSCRIBERS_ENDPOINT = `internal/patient/hl7v2-subscribers`;
+const csvSeparator = ",";
+const NUMBER_OF_PATIENTS_PER_PAGE = 500;
+const NUMBER_OF_ATTEMPTS = 3;
+const DELAY_BETWEEN_ATTEMPTS = dayjs.duration({ seconds: 1 });
```

Comment:
This isn't a pagination delay, i believe this is network retries - in which case let's make this something more like 30 / 60s.